src/lib.rs

```diff
@@ -32,9 +32,42 @@
 mod windows;
 
 use std::env;
+use std::ffi::OsString;
 use std::io;
 use std::path::{Path, PathBuf};
 
+/// Permits parameterising the home functions via the _from variants - used for
+/// in-process unit testing by rustup.
```

Comment:
If this trait is only used by rustup, which is in someway a internal implementation,
could we make it `#[doc(hidden)]`?

Reply:
I think it is potentially useful for anyone writing unit tests around this crate. But it is up to you. I wouldn't hide it.

There are equivalent things in other crates like clap - and those are not hidden.